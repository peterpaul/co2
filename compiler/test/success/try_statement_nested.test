package compiler.test;

["stdio.h"] int printf(char * format, ...);

class Exception
{
	char * message;
}

class AnException : Exception
{
}

int fun (Exception ex)
{
	if (ex != null)
	{
		throw ex;
	}
}

void loop (Exception ex)
{
	printf ("--------------------------------------------------------------------------------\n");
	try {
		printf ("loop.try\n");
		fun (ex);
	} catch (AnException ex) {
		printf ("loop.catch AnException\n");
		delete ex;
		throw new Exception ();
	} catch (Exception ex) {
		printf ("loop.catch Exception\n");
		delete ex;
	}
}

int main(int argc, char ** argv)
{
	try {
		loop (null);
		loop (new AnException());
		loop (new Exception());
	} catch (Exception ex) {
		printf ("main.catch Exception\n");
	}
	return 0;
}