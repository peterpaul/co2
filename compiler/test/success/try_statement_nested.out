--------------------------------------------------------------------------------
loop.try
--------------------------------------------------------------------------------
loop.try
loop.catch AnException
main.catch Exception
